snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -806,17 +876,22 @@ def prime(self, force=False) -> None:
     def _get_primed_stage_packages(self, snap_files: Set[str]) -> Set[str]:
         primed_stage_packages: Set[str] = set()
         for snap_file in snap_files:
-            snap_file = os.path.join(self.primedir, snap_file)
+            snap_file = os.path.join(self._project.prime_dir, snap_file)
             stage_package = xattrs.read_origin_stage_package(snap_file)
             if stage_package:
                 primed_stage_packages.add(stage_package)
         return primed_stage_packages
 
     def _do_prime(self) -> None:
         snap_files, snap_dirs = self.migratable_fileset_for(steps.PRIME)
-        _migrate_files(snap_files, snap_dirs, self.stagedir, self.primedir)
+        _migrate_files(
+            snap_files, snap_dirs, self._project.stage_dir, self._project.prime_dir
+        )
 
-        if self._snap_type == "app" and self._base is not None:
+        if (
+            self._project._snap_meta.type in ("app", None)
```

Comment:
is this an unrelated fix?  when would type be none?

Reply:
this is from the previous PR